src/main/java/org/datadog/jenkins/plugins/datadog/clients/ConcurrentMetricCounters.java

```diff
@@ -0,0 +1,72 @@
+package org.datadog.jenkins.plugins.datadog.clients;
+
+import net.sf.json.JSONArray;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.logging.Logger;
+
+public class ConcurrentMetricCounters {
+    private static final Logger logger = Logger.getLogger(ConcurrentMetricCounters.class.getName());
+    private static ConcurrentMetricCounters instance;
+    private static ConcurrentMap<CounterMetric, Integer> counters = new ConcurrentHashMap<>();
+
+    private ConcurrentMetricCounters(){}
+
+    public static ConcurrentMetricCounters getInstance(){
+        if(instance == null){
+            synchronized (ConcurrentMetricCounters.class) {
+                if(instance == null){
+                    instance = new ConcurrentMetricCounters();
+                }
+            }
+        }
+        return instance;
+    }
+
+    public static ConcurrentMap<CounterMetric, Integer> get(){
+        ConcurrentMetricCounters countersInstance = ConcurrentMetricCounters.getInstance();
+        return countersInstance.getCounters();
+    }
+
+    public static void reset(){
+        ConcurrentMetricCounters countersInstance = ConcurrentMetricCounters.getInstance();
+        countersInstance.resetCounters();
+    }
+
+    private void resetCounters(){
+        counters = new ConcurrentHashMap<>();
+    }
+
+    private ConcurrentMap<CounterMetric, Integer> getCounters(){
+        return counters;
+    }
```

Comment:
Are these really needed ?

Reply:
I prefer to use method rather than direct access. Not necessary but i prefer to keep it. We can remove later if it is really confusing.